sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/schema/JsonSchemaManager.java

```diff
@@ -0,0 +1,190 @@
+package io.stargate.sgv2.docsapi.service.schema;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.github.fge.jsonschema.core.exceptions.ProcessingException;
+import com.github.fge.jsonschema.core.report.ProcessingMessage;
+import com.github.fge.jsonschema.core.report.ProcessingReport;
+import com.github.fge.jsonschema.main.JsonSchemaFactory;
+import io.smallrye.mutiny.Uni;
+import io.stargate.bridge.proto.QueryOuterClass;
+import io.stargate.bridge.proto.Schema;
+import io.stargate.bridge.proto.StargateBridge;
+import io.stargate.sgv2.docsapi.api.common.StargateRequestInfo;
+import io.stargate.sgv2.docsapi.api.exception.ErrorCode;
+import io.stargate.sgv2.docsapi.api.exception.ErrorCodeRuntimeException;
+import io.stargate.sgv2.docsapi.service.schema.query.JsonSchemaQueryProvider;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+/**
+ * Json Schema manager provides basic operations to store, retrieve, and use the JSON schema for a
+ * Documents collection.
+ */
+@ApplicationScoped
+public class JsonSchemaManager {
+  @Inject ObjectMapper objectMapper;
+
+  @Inject JsonSchemaQueryProvider jsonSchemaQueryProvider;
+
+  @Inject StargateRequestInfo requestInfo;
+
+  private final JsonSchemaFactory jsonSchemaFactory = JsonSchemaFactory.byDefault();
+
+  /**
+   * Gets the JSON Schema for a given table.
+   *
+   * @param table
+   * @return Uni containing the JsonNode representing the schema
+   */
+  public Uni<JsonNode> getJsonSchema(Uni<Schema.CqlTable> table) {
+    // This properly handles authz based on which TableManager is provided
+    return table
+        .onItem()
+        .ifNotNull()
+        .transform(
+            t -> {
+              String comment = t.getOptionsMap().getOrDefault("comment", null);
+              if (comment == null) {
+                return null;
+              }
+
+              try {
+                return objectMapper.readTree(comment);
+              } catch (JsonProcessingException e) {
+                return null;
```

Comment:
maybe also a warn here for the logs, smth like `Document table has comment, but it's not a valid JSON schema`? With the exception of course..